xdsl/interactive/app.py

```diff
@@ -357,16 +386,7 @@ def watch_pass_pipeline(self) -> None:
         Function called when the reactive variable pass_pipeline changes - updates the
         label to display the respective generated query in the Label.
         """
-        self.selected_passes_list_view.clear()
-        for pass_value, value_spec in self.pass_pipeline:
-            self.selected_passes_list_view.append(
-                PassListItem(
-                    Label(pass_value.name),
-                    module_pass=pass_value,
-                    pass_spec=value_spec,
-                    name=pass_value.name,
-                )
-            )
+        self.update_selected_passes_list_view(self.pass_pipeline)
         self.update_current_module()
```

Comment:
I had a look into whether it's possible to have multiple watchers for the same reactive variable, and it seems a little tedious, so calling a few methods seems best.

Reply:
So this is okay?